express-notes-api/index.js

```diff
@@ -0,0 +1,92 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+const json = require('./data.json');
+
+
+app.use(express.json())
+
+const arr = []
+//sets up json
+for (let key in json.notes) {
+  arr.push({ content: json.notes[key].content, id: key })
+}
+
+
+app.get('/api/notes', (req, res) => {
+
+  res.status(200).send(JSON.stringify(arr, null, 2))
+  return
+});
+
+app.get('/api/notes/:id', (req, res) => {
+
+  if (Number(req.params.id) < 0 || isNaN(req.params.id) )  { //checks for negatives and NaNs
+    res.status(400).json({ error: 'The id must be a positive integer, and no strings' });
+  }
+  else if (Number(req.params.id) >= json.nextId) {
+    res.json({ error: `cannot find note with id ${req.params.id}` })
+  }
+  else if (Number(req.params.id) === json.notes[req.params.id].id) {
+    res.status(200).send(json.notes[req.params.id]) // will evaluate to a string type inside [brackets]
+  }
+});
+app.post('/api/notes', (req, res) => {
+  if (!req.body.content) { // no content means undefined aka falsy inside an if statement
+    res.status(400).send({ error: 'Content is a required field' });
+  } else {;
+    json.notes[json.nextId] = { content: req.body.content, id: json.nextId}; //directly puts an object as the value for a key, req and res still accessible
+    json.nextId++;
+    res.send(req.body)
```

Comment:
You cannot respond more than once to a request. I guarantee that your server terminal has a bunch of errors in it. `Cannot set headers after they are sent to the client`.